src/calendar-app/calendar/gui/EditCalendarDialog.ts

```diff
@@ -25,6 +27,83 @@ const defaultCalendarProperties: CalendarProperties = {
 	sourceUrl: "",
 }
 
+export async function handleUrlSubscription(calendarModel: CalendarModel, url: string): Promise<string | Error> {
+	if (!locator.logins.isFullyLoggedIn()) return new Error("notFullyLoggedIn_msg")
+
+	const externalIcalStr: string | Error = await calendarModel.fetchExternalCalendar(url).catch((e) => e as Error)
+	if (externalIcalStr instanceof Error || externalIcalStr.trim() === "") return new Error("fetchingExternalCalendar_error")
+
+	if (!isIcal(externalIcalStr)) return new Error("invalidICal_error")
+	return externalIcalStr
+}
+
+function sourceUrlInputField(urlStream: Stream<string>, errorMessageStream: Stream<string>) {
+	return m(TextField, {
+		value: urlStream(),
+		oninput: (url: string, inputElement: HTMLInputElement) => {
+			const assertionResult = checkURLString(url)
+			urlStream(url)
+			if (assertionResult instanceof URL) {
+				errorMessageStream("")
+				return
+			}
+			errorMessageStream(lang.get(assertionResult))
+		},
+		label: "url_label",
+		type: TextFieldType.Url,
+		helpLabel: () =>
+			m(
+				"small.block.content-fg",
+				urlStream().trim() === ""
+					? "E.g: https://mycalendarprovider.com/ics/calendar.ics"
```

Comment:
I don't think it's a good idea to have external links inside our app, mainly because we can't verify then.
If we go for a having the link, maybe we should use a more trustful source, e.g Thunderbird

Reply:
For now I left a clear example link using Tuta's domain. I don't think we should give providers recommendations without verifying them, at least not for now. If you think that's still unsuitable let me know.